subsys/llext/llext_link.c

```diff
@@ -479,6 +595,26 @@ int llext_link(struct llext_loader *ldr, struct llext *ext, const struct llext_l
 				return ret;
 			}
 
+			if (IS_ENABLED(CONFIG_RISCV)) {
+				/*
+				 * on RISC-V, for certain relocations, the arch_elf_relocate
+				 * function expects a value that depends on other relocations; we
+				 * resolve this in a helper function
+				 */
+				ret = llext_riscv_find_sym_pcrel(ext, ldr, &rel, shdr, &sym,
+								 &link_addr, &last_rel_idx);
+
+				if (ret != 0) {
+					LOG_ERR("Failed to resolve RISC-V PCREL relocation in "
+						"rela section %d entry %d!",
+						i, j);
+					return ret;
+				}
+			} else {
+				/* unused */
+				(void)last_rel_idx;
```

Comment:
does the compiler actually need this? Are there warnings without it?

Reply:
GCC would definitely complain if this was an ifdef rather than an if.